core/unix/loader.c

```diff
@@ -741,17 +741,55 @@ privload_search_rpath(privmod_t *mod, bool runpath, const char *name,
                     len = sep - list;
                 /* support $ORIGIN expansion to lib's current directory */
                 origin = strstr(list, RPATH_ORIGIN);
+#    ifdef CLIENT_INTERFACE
+                char path[MAXIMUM_PATH];
+#    endif
                 if (origin != NULL && origin < list + len) {
                     size_t pre_len = origin - list;
+#    ifdef CLIENT_INTERFACE
+                    snprintf(path, MAXIMUM_PATH, "%.*s%.*s%.*s", pre_len, list,
```

Comment:
Use BUFFER_SIZE_BYTES for less fragile code

Reply:
You mean s/MAXIMUM_PATH/BUFFER_SIZE_BYTES/? Do you want to change the existing filename snprintf as well?

Reply:
You can't change that one: it's not a static buffer.

Reply:
Done.